### PR TITLE
Bratseth/bcp rebased

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

File: config-model-api/src/main/java/com/yahoo/config/application/api/Bcp.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            return "BCP group of " + members;
+        }
+
+    }
+
+    public record RegionMember(RegionName region, double fraction) {
+
+        public RegionMember {
+            if (fraction < 0 || fraction > 1)
+                throw new IllegalArgumentException("Fraction must be a number between 0.0 and 1.0, but got " + fraction);
```
Comment (mpolden) [inline config-model-api/src/main/java/com/yahoo/config/application/api/Bcp.java:116]: Consider using `Validation.requireInRange`.

File: config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecWithBcpTest.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+                         <region>us-west2</region>
+                         <region>eu-east1</region>
+                         <region>eu-west1</region>
+                      </prod>
+                   </instance>
+                   <bcp>
+                     <group>
+                        <region>us-east1</region>
+                        <region>us-east2</region>
+                        <region fraction="0.3">us-central1</region>
```
Comment (mpolden) [inline config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecWithBcpTest.java:89]: I assume we'll discuss the syntax later. Maybe `traffic-share` or simply `share` would be more descriptive?

**Review by mpolden — APPROVED** [review]
Great!

File: config-model-api/src/main/java/com/yahoo/config/application/api/Bcp.java (hunk 1 above)
Comment (bratseth) [reply config-model-api/src/main/java/com/yahoo/config/application/api/Bcp.java:116]: Thanks for the tip. I'll do it in the next pr.

File: config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecWithBcpTest.java (hunk 2 above)
Comment (bratseth) [reply config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecWithBcpTest.java:89]: Yes, on Thursday. This isn't traffic share though, but fractional membership ...

**Pull request merged**